Sources/MapboxNavigation/NavigationView.swift

```diff
@@ -229,34 +236,194 @@ extension NavigationView {
         navigationMapView.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
         navigationMapView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
   
-        topBannerContainerView.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
-        topBannerContainerView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
-        topBannerContainerView.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        let regularTopBannerContainerViewLeadingConstraint = topBannerContainerView.leadingAnchor.constraint(equalTo: leadingAnchor)
+        let regularTopBannerContainerViewTrailingConstraint = topBannerContainerView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        let regularTopBannerContainerViewTopConstraint = topBannerContainerView.topAnchor.constraint(equalTo: topAnchor)
+        
+        let regularTopBannerConstraints = [
+            regularTopBannerContainerViewLeadingConstraint,
+            regularTopBannerContainerViewTrailingConstraint,
+            regularTopBannerContainerViewTopConstraint
+        ]
+        
+        let screenWidth = UIScreen.main.bounds.width
+        let compactTopBannerContainerViewLeadingConstraint = topBannerContainerView.leadingAnchor.constraint(equalTo: leadingAnchor)
+        let compactTopBannerContainerViewTrailingConstraint = topBannerContainerView.trailingAnchor.constraint(equalTo: trailingAnchor,
+                                                                                                               constant: -(screenWidth * 0.6))
```

Comment:
What does this look like when using guidance cards? That design has the upcoming card peek out from the trailing side, but now would it be clipped or float in the middle of the map somewhere?

Reply:
Yep, it's currently clipped. I'll have to improve layout there, but in general it also seem to look better (guidance card no longer fills the whole parent view width). For example:

![Simulator Screen Shot - iPhone 12 Pro Max - 2022-01-07 at 15 06 57](https://user-images.githubusercontent.com/1496498/148618634-4c00ffc8-0c24-4502-9b56-75174285aa6c.png)

Reply:
> guidance card no longer fills the whole parent view width

It’s certainly an improvement on an iPhone screen. Does this change affect the layout on iPad in either orientation? If I remember correctly, we had gotten it so that the subsequent cards would remain visible beside the current card, so that you’d see three up at a time in a sort of marquee layout. But maybe I’m confusing that with a custom integration with guidance cards at the application level.

Reply:
I've made a change to only show new layout in landscape mode for devices with regular and compact size classes, this means that iPads are not affected for now. I'm not sure that we should integrate new layout on iPads, as there is enough space in portrait/landscape mode anyway. Guidance cards width is also not affected, so only one guidance card is currently shown, next one is only partly visible. Current look:

![Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-01-20 at 11 18 19](https://user-images.githubusercontent.com/1496498/150407245-c1a5ed15-907f-477d-8c0f-908e72995ad7.png)

I'm thinking about doing some more tests to see whether we should perform any changes on iPad. Guidance cards improvements (which will be done in a separate ticket) might help with making a decision.